src/forms/TextInput.jsx

```diff
@@ -92,6 +94,10 @@ TextInput.propTypes = {
 		PropTypes.string,
 		PropTypes.element
 	]),
+	placeholder: PropTypes.oneOfType([
+		PropTypes.string,
+		PropTypes.element
```

Comment:
since placeholder is an html element attribute, it probably needs to be a real string end to end... hopefully not too difficult using `withIntl(Component, messages, true)` + `intl.formatMessage`